charmhelpers/core/sysctl.py

```diff
@@ -31,18 +31,22 @@
 def create(sysctl_dict, sysctl_file):
     """Creates a sysctl.conf file from a YAML associative array
 
-    :param sysctl_dict: a YAML-formatted string of sysctl options eg "{ 'kernel.max_pid': 1337 }"
+    :param sysctl_dict: a dict or YAML-formatted string of sysctl
+                        options eg "{ 'kernel.max_pid': 1337 }"
     :type sysctl_dict: str
     :param sysctl_file: path to the sysctl file to be saved
     :type sysctl_file: str or unicode
     :returns: None
     """
-    try:
-        sysctl_dict_parsed = yaml.safe_load(sysctl_dict)
-    except yaml.YAMLError:
-        log("Error parsing YAML sysctl_dict: {}".format(sysctl_dict),
-            level=ERROR)
-        return
+    if type(sysctl_dict) is not dict:
```

Comment:
This would probably be more flexible if it used:

```python
if isinstance(syctl_dict, dict):
```

as it would allow dict like objects to also be used.  If you wanted to be *really* flexible, then you could use:

```python
if isinstance(sysctl_dict, collections.abc.Mapping):
```

Then it would allow any object that supports the dictionary specific methods (e.g. __getattr__, __iter__, etc.) to be used.